app/helpers/api/v1/borrow_histories_helper.rb

```diff
@@ -0,0 +1,47 @@
+module Api::V1::BorrowHistoriesHelper
+
+  def display_date(date)
+    date.strftime('%Y-%m-%d')
+  end
+
+  # Custom validation method to check date range with accepted or done status
+  def check_date_range_with_accepted_or_done
+    @overlapping_borrows = BorrowHistory.where(book_id: book_id)
+                                      .where('(borrow_date, return_date) overlaps (?, ?)', borrow_date, return_date)
+                                      .where(status: [:accepted, :borrowed])
+    if @overlapping_borrows.exists?
+      errors.add(:base, 'Book already borrowed or accepted for the selected date range')
+    end
+
+  end
+
+  def borrow_date_cannot_be_in_past
+    if borrow_date.present? && borrow_date < Date.today
+      errors.add(:borrow_date, "can't be in the past ")
+    end
+  end
+
+  def schedule_notification_job
+    time_difference = (return_date - Time.zone.now).to_i
+    delay = time_difference > 0 ? time_difference : 0
+
+    NotificationWorkerJob.perform_in(delay, id)
+    end
+
+
+    def set_borrow_history
+      @borrow_history = BorrowHistory.find(params[:id])
```

Comment:
you are setting borrow history here correctly why not use it in the controller as a callback